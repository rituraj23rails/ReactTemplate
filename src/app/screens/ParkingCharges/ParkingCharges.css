.chargesMainContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.availableText {
    font-size: 20px;
    color: black;
}

.regisrationText {
    font-size: 30px;
    color: blue;
    margin-top: 20px;
}




