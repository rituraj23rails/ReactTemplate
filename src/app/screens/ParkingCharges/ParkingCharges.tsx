import React, { Component, useRef } from 'react';
import './ParkingCharges.css';
import WithRouter from '../../../WithRouter';
import { connect } from 'react-redux';
import { updateParkingData } from '../ParkingSlots/ParkingSlice';

export interface Props {
    navigate:  any;
    id: string;
    location: any;
    updateParkingData: (parkingData: any) => void;
    parkingData: Array<any>;
}

interface S {
    slotData: any;
    totalHrs: number;
    totalCharges: number;
}

interface SS {
    id: any;
}

class ParkingCharges extends Component<Props, S, SS> {
    carousel: any = null;
    constructor(props: any) {
        super(props);
        this.state = {
            slotData: this.props.location.state.slotData,
            totalCharges: 0,
            totalHrs: 0,
        };
    };

    componentDidMount = () => {
        let slotData = this.state.slotData;
        let currentTime = new Date().getTime();
        let totalHrs = Math.abs(Math.floor((currentTime - slotData.time)/(1000 * 60*60)));
        let totalCharges = 0;
        if(totalHrs <= 2) {
            totalCharges = 10;
        } else {
            totalCharges = 10 + 10 * (totalHrs - 2);
        }
        this.setState({ totalHrs: totalHrs, totalCharges: totalCharges });
    }

    componentWillUnmount = () => {

    }


    onExitCarRegistration = () => {
        let availableParkingArea = JSON.parse(JSON.stringify(this.props.parkingData));
        let availableIndex = availableParkingArea.findIndex((carItem: any) => carItem.uniqueNo === this.state.slotData.uniqueNo);
        availableParkingArea[availableIndex]['isAvailable'] = true;
        this.props.updateParkingData(availableParkingArea);
        alert('Payment Done.')
        this.props.navigate(-1);
    }

    render() {
        return (
            <div className='chargesMainContainer'>
                <p className="regisrationText">Car Registration No: {this.state.slotData.carRegistrationNo}</p>
                <p className="regisrationText">Total Time: {this.state.totalHrs} hrs.</p>
                <p className="regisrationText">Total Charges: ${this.state.totalCharges}</p>
                <button data-testid="submitButtonTestID" onClick={() => this.onExitCarRegistration()} className="submitButton">DEALLOCATE CAR</button>
            </div>
        )
    }
};

const mapStateToProps = (state: any) => ({
    parkingData: state.parking.parkingData
});

const mapDispatchToProps = {
    updateParkingData: updateParkingData
};

export default connect(mapStateToProps, mapDispatchToProps)(WithRouter(ParkingCharges));
export {ParkingCharges};
