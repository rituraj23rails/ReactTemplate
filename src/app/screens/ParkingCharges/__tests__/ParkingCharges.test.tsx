/**
 *
 PARKING HOME TEST FILE
 * @format
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper, configure, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { ParkingCharges } from '../ParkingCharges';

configure({ adapter: new Adapter() });


const screenProps = {
    navigate: jest.fn(),
    id: 'ParkingCharges',
    parkingData: [{
        isAvailable: false,
        carRegistrationNo: '111',
        uniqueNo: 1
    }],
    location: {
        state: {
            slotData: {
                isAvailable: false,
                carRegistrationNo: '111',
                uniqueNo: 1,
                time: 1
            }
        }
    },
    updateParkingData: jest.fn()
}

describe('Launching ParkingCharges', () => {
    
    beforeEach(() => {
        global.fetch = jest.fn()
        window.alert = jest.fn();
    })

    it('ParkingCharges root file loads properly', () => {
        const tree: any = renderer.create(<ParkingCharges {...screenProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Test User ParkingCharges Empty Data Screen', () => {
        let wrapper: ShallowWrapper;
        let instance: ParkingCharges;
        wrapper = shallow(<ParkingCharges {...screenProps} />)
        instance = wrapper.instance() as ParkingCharges;
        let buttonComponent = wrapper.findWhere((node) => node.prop("data-testid") === "submitButtonTestID");
        buttonComponent.simulate("click");
        instance.componentWillUnmount();
    });
});