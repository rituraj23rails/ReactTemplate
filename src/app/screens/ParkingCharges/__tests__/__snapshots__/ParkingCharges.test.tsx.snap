// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launching ParkingCharges ParkingCharges root file loads properly 1`] = `
<div
  className="chargesMainContainer"
>
  <p
    className="regisrationText"
  >
    Car Registration No: 
    111
  </p>
  <p
    className="regisrationText"
  >
    Total Time: 
    459578
     hrs.
  </p>
  <p
    className="regisrationText"
  >
    Total Charges: $
    4595770
  </p>
  <button
    className="submitButton"
    data-testid="submitButtonTestID"
    onClick={[Function]}
  >
    DEALLOCATE CAR
  </button>
</div>
`;
